apps/api/v2/src/modules/auth/guards/organization/organization-id.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { OrganizationsRepository } from "@/modules/organizations/organizations.repository";
+import { Injectable, CanActivate, ExecutionContext, ForbiddenException } from "@nestjs/common";
+
+@Injectable()
+export class OrganizationIdGuard implements CanActivate {
```

Comment:
IMO the check whether organization is actually an organization not a team should be handled in the individual functions e.g. `OrganizationUsersRepository.getOrganizationUsers` below or "isOrganization: true" should be included in the query itself.

It can be a separate function in the repository that acts as a railguard for all repository calls. I am saying this because organizations service and organizations repository in the future can also be dependency injected into other parts of the api, and then that would mean they would also have to remember using this guard.

Reply:
it makes sense to have a guard used for endpoints that have :orgId as a dynamic param, I agree with Lauris to have the logic to throw a specific error when an organization does not exist directly within the repository.